cmd/soroban-cli/src/commands/contract/deploy/wasm.rs

```diff
@@ -202,21 +195,17 @@ impl NetworkRunnable for Cmd {
         client
             .verify_network_passphrase(Some(&network.network_passphrase))
             .await?;
-        let key = config.key_pair()?;
-
         // Get the account sequence number
-        let public_strkey =
-            stellar_strkey::ed25519::PublicKey(key.verifying_key().to_bytes()).to_string();
-
-        let account_details = client.get_account(&public_strkey).await?;
+        let public_strkey = config.public_key().await?;
+        let account_details = client.get_account(&public_strkey.to_string()).await?;
         let sequence: i64 = account_details.seq_num.into();
         let (txn, contract_id) = build_create_contract_tx(
             wasm_hash,
             sequence + 1,
             self.fee.fee,
             &network.network_passphrase,
             salt,
-            &key,
+            &public_strkey,
         )?;
         if self.fee.build_only {
```

Comment:
Hmmm it looks like currently --`build-only` and --`sim-only` actually conflict with` --sign-with`, because `--sign-with` is getting effectively ignored. I think it makes sense that we always sign the transaction, but we could also make sign-with flags to conflict with `--build-only` and `--sim-only`

Reply:
For --build-only I think we should disallow signing with --sign-with because the tx isn't ready to sign and so we'd just be setting up a user for pain.

For --sim-only the txn is ready to sign, so having --sim-only and then a signer is fine I think, the outputted txn will be ready to submit.

Reply:
Given that we can `stellar ... --sim-only | stellar tx sign` perhaps they should be mutually exclusive as it's a little surprising that it's doing more than signing. We could print a warning if the options are used together.

Regardless I would prefer this in a future PR as it would involve touching each early return for all commands that have `--sim-only`.

Reply:
Agree, let's do a separate PR, just a warning for now should be good

Reply:
👍🏻 I think for v22 we should also remove the `--sim-only`. There's multiple ways to achieve it because of the `tx simulate subcommand, and if we remove `--sim-only` it's one less variants of use we need to maintain.

Reply:
Opened this issue to follow up with removing --sim-only closer to v22:
- https://github.com/stellar/stellar-cli/issues/1547

Reply:
@leighmcculloch so we want people to: 
`stellar contract <command> | stellar tx simulate | stellar tx sign --withKey alice | stellar tx sign --withKey bob` 
right?
Strongly agree on cutting ways of doing the same thing running different commands